radixdlt/src/main/java/com/radixdlt/consensus/ValidatingEventCoordinator.java

```diff
@@ -235,12 +235,11 @@ public void processProposal(Vertex proposedVertex) {
 			log.error(this.getShortName() + ": PROPOSAL: Rejected " + proposedVertex, e);
 		}
 
-		// If currently node leader, Proceed to next view
-		if (!Objects.equals(proposerElection.getProposer(updatedView), selfKey.getPublicKey())) {
+		// If not currently leader or next leader, Proceed to next view
+		if (!Objects.equals(proposerElection.getProposer(updatedView), selfKey.getPublicKey())
+			&& !Objects.equals(proposerElection.getProposer(updatedView.next()), selfKey.getPublicKey())) {
```

Comment:
Why does the next leader not advance to the next view here?

Reply:
since we're following chained hot stuff at the moment, we need to wait for the previous leader's QC to guarantee we have the latest QC. See the Chained HotStuff pseudocode in the paper.

Reply:
Ah right, since we changed back from event-driven